src/Interpreters/Aggregator.h

```diff
@@ -209,7 +209,8 @@ struct AggregationMethodOneNumber
     // Insert the key from the hash table into columns.
     static void insertKeyIntoColumns(const Key & key, std::vector<IColumn *> & key_columns, const Sizes & /*key_sizes*/)
     {
-        const auto * key_holder = reinterpret_cast<const char *>(&key);
+        FieldType casted_key = static_cast<FieldType>(key);
+        const auto * key_holder = reinterpret_cast<const char *>(&casted_key);
         auto * column = static_cast<ColumnVectorHelper *>(key_columns[0]);
         column->insertRawData<sizeof(FieldType)>(key_holder);
```

Comment:
maybe something like this?
```suggestion
        static_assert(sizeof(FieldType) <= sizeof(Key));
        const auto * key_holder = reinterpret_cast<const char *>(&key);
        auto * column = static_cast<ColumnVectorHelper *>(key_columns[0]);
        if constexpr (sizeof(FieldType) < sizeof(Key) && std::endian::native == std::endian::little)
            column->insertRawData<sizeof(FieldType)>(key_holder + (sizeof(Key) - sizeof(FieldType)));
        else
            column->insertRawData<sizeof(FieldType)>(key_holder);
```

Reply:
The loss happens before that, what about something like this?
```suggestion
        static_assert(sizeof(FieldType) <= sizeof(Key));
        auto * column = static_cast<ColumnVectorHelper *>(key_columns[0]);
        if constexpr (sizeof(FieldType) < sizeof(Key) && std::endian::native == std::endian::little)
        {
            const auto * key_holder = reinterpret_cast<const char *>(&key);
            column->insertRawData<sizeof(FieldType)>(key_holder + (sizeof(Key) - sizeof(FieldType)));
        }
        else
        {
            auto casted_key = static_cast<FieldType>(key);
            const auto * key_holder = reinterpret_cast<const char *>(&casted_key);
            column->insertRawData<sizeof(FieldType)>(key_holder);
        }
```

Reply:
sorry, I made a mistake in my suggested change - the block with offset was meant for big-endian - I fixed it, look again. The idea is that `insertRawData` is using `sizeof(FieldType)` bytes, so for small keys, when sizeof(FieldType) < sizeof(Key), on big-endian it will take bytes with zeroes - suggested change allows to offset taken bytes to those having value.

Reply:
Thanks, functional test passed on s390x platform.